File: ios/SenderKeyRefs.swift
import Foundation
import ExpoModulesCore
import LibSignalClient

final class SenderKeyRecordRef: SharedObject {
  let record: SenderKeyRecord

  init(record: SenderKeyRecord) {
    self.record = record
    super.init()
  }
}

final class SenderKeyDistributionMessageRef: SharedObject {
  let message: SenderKeyDistributionMessage

  init(message: SenderKeyDistributionMessage) {
    self.message = message
    super.init()
  }
}

File: android/src/main/java/expo/modules/libsignal/SenderKeyRefs.kt
package expo.modules.libsignal

import expo.modules.kotlin.sharedobjects.SharedObject
import org.signal.libsignal.protocol.groups.state.SenderKeyRecord
import org.signal.libsignal.protocol.message.SenderKeyDistributionMessage

class SenderKeyRecordRef(val record: SenderKeyRecord) : SharedObject()

class SenderKeyDistributionMessageRef(val message: SenderKeyDistributionMessage) : SharedObject()
